packages/core/src/types/Matrix2D.ts

```diff
@@ -0,0 +1,663 @@
+import {EPSILON, PossibleMatrix2D} from './Matrix';
+import {PossibleVector2, Vector2} from './Vector';
+
+/**
+ * A specialized 2x3 Matrix representing a 2D transformation.
+ *
+ * A Matrix2D contains six elements defined as
+ * [a, b,
+ *  c, d,
+ *  tx, ty]
+ *
+ * This is a shortcut for a 3x3 matrix of the form
+ * [a, b, 0,
+ *  c, d, 0
+ *  tx, ty, 1]
+ *
+ * Note that because a Matrix2D ignores the z-values of each component vectors,
+ * it does not satisfy all properties of a "real" 3x3 matrix.
+ *
+ *   - A Matrix2D has no transpose
+ *   - A(B + C) = AB + AC does not hold for a Matrix2D
+ *   - (rA)^-1 = r^-1 A^-1, r != 0 does not hold for a Matrix2D
+ *   - r(AB) = (rA)B = A(rB) does not hold for a Matrix2D
+ */
+export class Matrix2D {
+  public readonly values: Float32Array = new Float32Array(6);
+
+  public static get identity(): Matrix2D {
+    return new Matrix2D(1, 0, 0, 1, 0, 0);
+  }
```

Comment:
This could be a constant:
```suggestion
  public static readonly identity = new Matrix2D(1, 0, 0, 1, 0, 0);
```

Reply:
That won't work if we also want to include the setters as that would mean you could do

```ts
Matrix2D.identity.translation = [100, 100];

Matrix.identity.translation;
// => Vector2(100, 100)
```

Reply:
Yes, but the types are meant to be immutable. If you mutate a type it means that you know what you're doing.
It should be:
```ts
Matrix2D.identity.translate(100, 100)
```
We have the same thing going on in `Vector2`